app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -112,42 +117,93 @@ class PopupView : FrameLayout {
             return ViewHolder(binding)
         }
 
-        override fun onBindViewHolder(holder: ViewHolder, position: Int) =
-            holder.bind(getItem(position))
+        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+            val popupField: PopupField = getItem(position)
+            holder.updateViewMode(popupField)
+
+            holder.bind(popupField)
+        }
     }
 
     /**
      * Callback for calculating the diff between two non-null items in a list.
      */
-    private class DiffCallback : DiffUtil.ItemCallback<PopupAttribute>() {
+    private class DiffCallback : DiffUtil.ItemCallback<PopupField>() {
 
         override fun areItemsTheSame(
-            oldItem: PopupAttribute,
-            newItem: PopupAttribute
+            oldItem: PopupField,
+            newItem: PopupField
         ): Boolean {
             return oldItem == newItem
         }
 
         override fun areContentsTheSame(
-            oldItem: PopupAttribute,
-            newItem: PopupAttribute
+            oldItem: PopupField,
+            newItem: PopupField
         ): Boolean {
-            return oldItem.field == newItem.field && oldItem.value == newItem.value
+            return oldItem.fieldName == newItem.fieldName
```

Comment:
I think you should use String's equals method here.

```suggestion
            return oldItem.fieldName.equals(newItem.fieldName)
```

Reply:
The "==" operator in Kotlin works different than Java, it checks the contents for comparison. 

https://kotlinlang.org/docs/reference/equality.html